foundry/pixi/containers/canvasLayers/placeablesLayers/wallsLayer.d.ts

```diff
@@ -0,0 +1,241 @@
+
+  /**
+   * Get the endpoint coordinates for a wall placement, snapping to grid at a specified precision
+   * Require snap-to-grid until a redesign of the wall chaining system can occur.
+   * @param  point - The initial candidate point
+   * @param  snap  - Whether to snap to grid
+   *                 (default: `true`)
+   * @returns The endpoint coordinates [x,y]
+   */
+  protected _getWallEndpointCoordinates(point: Point, { snap }?: { snap?: boolean }): PointArray;
+
+  /**
+   * The Scene Controls tools provide several different types of prototypical Walls to choose from
+   * This method helps to translate each tool into a default wall data configuration for that type
+   * @param tool - The active canvas tool
+   */
+  protected _getWallDataFromActiveTool(
+    tool: string
+  ):
+    | {
+        move: ValueOf<typeof CONST['WALL_MOVEMENT_TYPES']>;
+        sense: ValueOf<typeof CONST['WALL_SENSE_TYPES']>;
+        door: ValueOf<typeof CONST['WALL_DOOR_TYPES']>;
+      }
+    | Wall['data'];
+
+  /** @override */
+  protected _onClickLeft(event: PIXI.InteractionEvent): void;
+
+  /** @override */
+  protected _onDragLeftStart(event: PIXI.InteractionEvent): void;
+
+  /** @override */
+  protected _onDragLeftMove(event: PIXI.InteractionEvent): void;
+
+  /** @override */
+  protected _onDragLeftDrop(event: PIXI.InteractionEvent): void;
+
+  /** @override */
+  protected _onDragLeftCancel(event: PointerEvent): void;
+
+  /** @override */
+  protected _onClickRight(event: PIXI.InteractionEvent): void;
+
+  /**
+   * Test a single Ray against a single Wall
+   * @param ray  - The Ray being cast
+   * @param wall - The Wall against which to test
+   * @returns An intersection, if one occurred
+   */
+  static testWall(ray: Ray, wall: Wall): Vector2 | null;
+
+  /**
+   * Identify the closest collision point from an array of collisions
+   * @param collisions - An array of intersection points
+   * @returns The closest blocking intersection
+   */
+  static getClosestCollision(collisions: RayIntersection[]): RayIntersection | null;
+
+  /**
+   * Get the set of wall collisions for a given Ray
+   * @param ray           - The Ray being tested
+   * @param blockMovement - Test against walls which block movement?
+   * @param blockSenses   - Test against walls which block senses?
+   * @param mode          - The return mode of the test, one of "all", "closest", or "any"
+   * @param _performance  - An internal performance object used for debugging
+   * @returns An array of collisions, if mode is "all"; The closest collision, if mode is "closest"; Whether any collision occurred if mode is "any"
+   */
+  static getRayCollisions(
+    ray: Ray,
+    {
+      blockMovement,
+      blockSenses,
+      mode
+    }?: { blockMovement: boolean; blockSenses: boolean; mode: 'all' | 'closest' | 'any'; _performance: unknown }
+  ):
+    | Record<`${string},${string}`, { x: number; y: number; t0: number; t1: number }>[]
```

Comment:
the type with {x: number, ...}` is just `Vector2`, right? Please also coordinate here to get this cleaned up when bot this and #534 are being merged.